internal/commands/push/command_test.go

```diff
@@ -1027,6 +1027,47 @@ Try instead: realm-cli push --local testdata/dependencies --remote appID --inclu
 
 		assert.Nil(t, err)
 	})
+
+	t.Run("with local path in nested directory", func(t *testing.T) {
+		var realmClient mock.RealmClient
+		realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+			return []realm.App{{GroupID: "groupID"}}, nil
+		}
+		realmClient.CreateAppFn = func(groupID, name string, meta realm.AppMeta) (realm.App, error) {
+			return realm.App{
+				GroupID: groupID,
+				ID:      primitive.NewObjectID().Hex(),
+				Name:    name,
+				AppMeta: meta,
+			}, nil
+		}
+		realmClient.DiffFn = func(groupID, appID string, appData interface{}) ([]string, error) {
+			return []string{}, nil
+		}
+
+		out := new(bytes.Buffer)
+		ui := mock.NewUIWithOptions(mock.UIOptions{AutoConfirm: true}, out)
+
+		cmd := &Command{inputs{LocalPath: filepath.Join(testApp.RootDir, "functions"), RemoteApp: "appID"}}
+		err := cmd.Handler(nil, ui, cli.Clients{Realm: realmClient})
+		defer os.Remove(filepath.Join(testApp.RootDir, local.FileRealmConfig.String()))
```

Comment:
Yeah, I saw this... The problem we have here is a problem in `createNewApp`. [This line](https://github.com/10gen/realm-cli/blob/master/internal/commands/push/command.go#L478) creates a new config file based on the default app config version, which means if the CLI can't resolve an app it's always going to create a new config file.

Options I can think of here are:
- Updating `createNewApp` to write to the "correct" config file (i.e. if local has `stitch.json`, write to `stitch.json`). I'm guessing this might lead to compatibility issues but I'm not sure.
- Bump the project version for this test suite and use `realm_config.json`. I don't think the suite is too dependent on the old project verison so this oculd be a quick and easy way to set this up.
- Set up this test in a temp directory which we can clean up after the test finishes. Also a good solution but would involve dynamically generating a `realm_config.json`
- Mock the call to `command.createNewApp` altogether. This is the cleanest, best solution IMO, but I'm not sure the CLI repo has a library for mocking/spying yet.

Reply:
> Bump the project version for this test suite and use realm_config.json. I don't think the suite is too dependent on the old project version so this could be a quick and easy fix.

We still want to support old cli versions -- we should actually add tests for each 

> Updating createNewApp to write to the "correct" config file (i.e. if local has stitch.json, write to stitch.json). I'm guessing this might lead to compatibility issues but I'm not sure.

This seems like the appropriate solution to me. Ideally, if the user has a config file they are trying to push, we would keep the version they're using. 
Maybe @makesitgo can chime in here.

Reply:
so my first though is: perhaps we make a new top-level test func for this: `TestPushCommandCreateNewAppNested(t *testing.T)` (there's a lot in this test file already... there's no harm in cornering off a little section of our own for this fix)

as far as cleanup/teardown goes, check out the test table's implementation found in `TestPushCommandCreateNewApp` (around L#1150 for my local `master`).  something tells me we can test the push command handler for your fix here by using `mock.NewProfileFromTmpDir(...)` (the usages of that function may prove useful to look at as well)

utlimately, I don't think we need to do what `TestPushCommandCreateNewApp` is doing and make our new tests "interactive"... but I do think it'd be worthwhile to sort of follow what `TestPushHandler` and run this flow in a more "end-to-end" fashion by executing `*Command.Handler` (and specify `AutoConfirm`)

I like jamie's thoughts on running this test for all config versions, that'd be a good thing to try and achieve here

---

also as a complete aside, we don't really need any library or package for mocking/spying things in `go`... we can just write some oddly-syntaxed things:

```go
// instead of
func createNewApp(...) {
  // func impl here
}

// you could write
var createNewApp = func(...) {
  // same func impl here
}

// the funcs can then be consumed/used the same way

// but in a test, you have a change to replace it
func TestSomething(t *testing.T) {
  origCreateNewApp := createNewApp
  createNewApp = func(...) {
    // some mock or spy impl
  }
  // and in this test/mock pattern, it's always good to remember to cleanup afterwards
  defer func() { createNewApp = origCreateNewApp }()
}
```